pkg/shell.go

```diff
@@ -20,3 +23,82 @@ func ExecShellReturnStrings(command string, args ...string) (string, string, err
 	log.Println("Commad Execution STDERR: %s", errb.String())
 	return outb.String(), errb.String(), err
 }
+
+func ExecShellReturnStringsWithVars(osvars map[string]string, command string, args ...string) (string, string, error) {
+	
+	log.Printf("INFO: Running %s",command)
+	for k, v := range osvars {		
+		os.Setenv(k, v)
+		log.Printf(" export %s = %s", k, v)
+	}
+	var outb, errb bytes.Buffer
+	k := exec.Command(command, args...)
+	k.Stdout = &outb
+	k.Stderr = &errb
+	err := k.Run()
+	if err != nil {
+		log.Println("Error executing command: %v", err)
+	}
+	log.Println("Commad Execution: %s", command)
+	log.Println("Commad Execution STDOUT: %s", outb.String())
+	log.Println("Commad Execution STDERR: %s", errb.String())
+	return outb.String(), errb.String(), err
+}
+
+// Exec shell actions supporting:
+// - On-the-fly logging of result
+// - Map of Vars loaded
+func ExecShellWithVars(osvars map[string]string, command string, args ...string) (error) {
```

Comment:
one pattern that I have been trying to follow, and I believe it's nice for developers, is to add a longer description on what the functions is doing, like:
`// ExecShellWithVars does this and that`

it will be beneficial on the long run at the moment we want to export the code documentation to godocs, example: https://pkg.go.dev/io#ByteScanner

Reply:
Not sure what to do yet, I would like to setup some godocs, but there is not much fancy stuff to be done there as fare I found.  This PR is more to improve the lops to not break the progress bars.